pnpm-lock.yaml

```diff
@@ -10530,7 +10501,7 @@ packages:
       object-assign: 4.1.1
       vary: 1.1.2
 
-  /cosmiconfig-typescript-loader@4.3.0(@types/node@18.11.18)(cosmiconfig@8.0.0)(ts-node@10.9.1)(typescript@4.9.4):
+  /cosmiconfig-typescript-loader@4.3.0(@types/node@14.18.36)(cosmiconfig@8.0.0)(ts-node@10.9.1)(typescript@4.9.4):
```

Comment:
Users running `pnpm install` on an earlier version of pnpm might revert the preferred resolutions switched by `pnpm dedupe` from a later version of pnpm. We've noticed this flavor of problem cause confusion in PRs, when different versions of pnpm have slightly different dependency preferences and create unexpected `pnpm-lock.yaml` changes that are committed.

Just a heads up that differences in `resolveDependencies` logic across the same major version of pnpm can be surprising.

Reply:
this affects only the deduping optimization, so it shouldn't break any user, so I wouldn't expect a major version for such change (maybe a minor)

Reply:
It is already an issue as you can see. It randomly updates the versions of `@types/node` even though I did not touch `@types/node`. I am not sure even that this PR will fix all the issues related to this. Maybe we'll have to have another look at how preferred versions are selected.